src/derivers/bip32.ts

```diff
@@ -166,22 +172,20 @@ async function deriveSecretExtension({
 }: DeriveSecretExtensionArgs) {
   if (isHardened) {
     // Hardened child
-    const indexBuffer = Buffer.allocUnsafe(4);
-    indexBuffer.writeUInt32BE(childIndex + BIP_32_HARDENED_OFFSET, 0);
-    const pk = privateKey;
-    const zb = Buffer.alloc(1, 0);
-    return Buffer.concat([zb, pk, indexBuffer]);
+    const indexBytes = new Uint8Array(4);
+    const view = new DataView(indexBytes.buffer);
+
+    view.setUint32(0, childIndex + BIP_32_HARDENED_OFFSET, false);
+    return concatBytes([new Uint8Array([0]), privateKey, indexBytes]);
   }
 
   // Normal child
-  const indexBuffer = Buffer.allocUnsafe(4);
-  indexBuffer.writeUInt32BE(childIndex, 0);
   const parentPublicKey = await curve.getPublicKey(privateKey, true);
-  return Buffer.concat([parentPublicKey, indexBuffer]);
+  return derivePublicExtension({ parentPublicKey, childIndex, curve });
```

Comment:
Why is the curve included here?

Reply:
It's part of the type. Might have been used at some point, but I can remove it now, I guess.

Reply:
Are you going to remove this?

Reply:
Done!